test/unit/excel/test_loading.py

```diff
@@ -0,0 +1,90 @@
+
+        self.assertEqual(study_entity1, study_entity2)
+        self.assertNotEqual(sample_entity1, sample_entity2)
+
+    def test_loading_index_equality(self):
+        submission = ExcelSubmission()
+        study_entity1 = ExcelLoader.map_row_entity(submission, 1, 'study', {'study_alias': 'STUD1'})
+        sample_entity1 = ExcelLoader.map_row_entity(submission, 1, 'sample', {'sample_index': 'SAME1'})
+        run_entity1 = ExcelLoader.map_row_entity(submission, 1, 'run', {'run_name': 'John'})
+
+        study_entity2 = ExcelLoader.map_row_entity(submission, 2, 'study', {'study_alias': 'STUD1'})
+        sample_entity2 = ExcelLoader.map_row_entity(submission, 2, 'sample', {'sample_index': 'SAME1'})
+        run_entity2 = ExcelLoader.map_row_entity(submission, 2, 'run', {'run_name': 'Jill'})
+
+        self.assertEqual(study_entity1, study_entity2)
+        self.assertEqual(sample_entity1, sample_entity2)
+        self.assertNotEqual(run_entity1, run_entity2)
+    
+    def test_loading_without_index(self):
+        submission = ExcelSubmission()
+        entity1 = ExcelLoader.map_row_entity(submission, 1, 'lorem', {})
+        entity2 = ExcelLoader.map_row_entity(submission, 2, 'ipsum', {})
+
+        self.assertNotEqual(entity1, entity2)
+        self.assertEqual('lorem:1', entity1.identifier.index)
+        self.assertEqual('ipsum:2', entity2.identifier.index)
+    
+    def test_links(self):
+        submission = ExcelSubmission()
+        study_entity1 = ExcelLoader.map_row_entity(submission, 1, 'study', {'study_accession': 'STUD1'})
+        sample_entity1 = ExcelLoader.map_row_entity(submission, 1, 'sample', {'sample_accession': 'SAME1'})
+        run_entity1 = ExcelLoader.map_row_entity(submission, 1, 'run_experiment', {'run_experiment_accession': 'RUN1'})
+        ExcelLoader.map_row_entity(submission, 2, 'study', {'study_accession': 'STUD1'})
+        ExcelLoader.map_row_entity(submission, 2, 'sample', {'sample_accession': 'SAME1'})
+        run_entity2 = ExcelLoader.map_row_entity(submission, 2, 'run_experiment', {'run_experiment_accession': 'RUN2'})
+        
+        self.assertSetEqual({'SAME1'}, study_entity1.get_linked_indexes('sample'))
+        self.assertSetEqual({'RUN1', 'RUN2'}, study_entity1.get_linked_indexes('run_experiment'))
+
+        self.assertSetEqual({'STUD1'}, sample_entity1.get_linked_indexes('study'))
+        self.assertSetEqual({'RUN1', 'RUN2'}, sample_entity1.get_linked_indexes('run_experiment'))
+
+        self.assertSetEqual({'STUD1'}, run_entity1.get_linked_indexes('study'))
+        self.assertSetEqual({'SAME1'}, run_entity1.get_linked_indexes('sample'))
+
+        self.assertSetEqual({'STUD1'}, run_entity2.get_linked_indexes('study'))
+        self.assertSetEqual({'SAME1'}, run_entity2.get_linked_indexes('sample'))
+
+    def test_loading_differing_index_equality(self):
+        # Note: I'm not sure if this is a feature or a bug so if this test fails it's okay,
```

Comment:
Well, to me this seems like a bug

Reply:
I've highlighted a problem with our current design, it won't be experienced by any of our users, and I don't think we should stop this merge to fix this one issue.
Would you me to create this as a GitHub issue so we can get back to it at some point?

Reply:
I did not want to stop this merge because of this issue. I think we should carefully investigate this issue and fix it if it looks like a bug.
Yes, we have to create a ticket on Trello. I guess you meant Trello and not GitHub. ;-)